glean-core/android/build.gradle

```diff
@@ -122,8 +129,9 @@ dependencies {
     // avoiding other configurations from being resolved.  Tricky!
     testImplementation files(configurations.jnaForTest.copyRecursive().files)
     testImplementation 'junit:junit:4.12'
-    testImplementation 'org.robolectric:robolectric:3.8'
+    testImplementation 'org.robolectric:robolectric:4.2'
```

Comment:
Let's use the same versions as the ones on a-c, since the latest ones have problems with dealing with Windows paths :E (a-c is using 4.1, see [here](https://github.com/mozilla-mobile/android-components/blob/63f5be14baa404af58caea9080a7101aaef1af4a/buildSrc/src/main/java/Dependencies.kt#L11))